### PR TITLE
Manage api constants immutable

### PR DESCRIPTION
**Pull request by scottf**
- management api read with examples
- ApiConstants to share filed name constants and Json patterns
- api objects made immutable
- reusable JsonUtils methods to read fields to simplify construction and support immutability
- ConsumerConfiguration maxWaiting not a field.
- Verified default field values according to jetstream/api/v1 some -1 changes to 0, max_waiting removed, addresses TODOs

### CODE REVIEW

File: src/examples/java/io/nats/examples/NatsJsManageStreams.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            System.out.println("----------\n5. Purge stream");
+            PurgeResponse purgeResponse = jsm.purgeStream(STREAM1);
+            printObject(purgeResponse);
+
+            // 6.1 Delete a stream
+            // 6.2 getStreamInfo should exception
+            // 6.3 deleteStream should exception
+            // 6.4 purgeStream should exception
+            System.out.println("----------\n6. Delete stream");
+            jsm.deleteStream(STREAM2);
```
Comment (ColinSullivan1) [inline src/examples/java/io/nats/examples/NatsJsManageStreams.java:117]: I'm not sure the calls below are required for the example, the APIs are demonstrated above - I'd think the comment above noting that subsequent calls will throw an exception should suffice.

File: src/main/java/io/nats/client/impl/PurgeResponse.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    private final boolean success;
+    private final int purged;
+
+    public PurgeResponse(String json) {
+        success = JsonUtils.readBoolean(json, SUCCESS_RE);
+        purged = JsonUtils.readInt(json, PURGED_RE, 0);
+    }
+
+    public boolean isSuccess() {
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/impl/PurgeResponse.java:30]: Needs comments....

File: src/main/java/io/nats/client/impl/PurgeResponse.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.regex.Pattern;
+
+public class PurgeResponse {
+    private static final Pattern SUCCESS_RE = JsonUtils.buildPattern("success", JsonUtils.FieldType.jsonBoolean);
+    private static final Pattern PURGED_RE = JsonUtils.buildPattern("purged", JsonUtils.FieldType.jsonNumber);
+
+    private final boolean success;
+    private final int purged;
+
+    public PurgeResponse(String json) {
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/impl/PurgeResponse.java:25]: Does this need to be public?

File: src/main/java/io/nats/client/impl/PurgeResponse.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        success = JsonUtils.readBoolean(json, SUCCESS_RE);
+        purged = JsonUtils.readInt(json, PURGED_RE, 0);
+    }
+
+    public boolean isSuccess() {
+        return success;
+    }
+
+    public int getPurged() {
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/impl/PurgeResponse.java:34]: Needs comments

File: src/main/java/io/nats/client/StreamState.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
+package io.nats.client;
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/StreamState.java:1]: License/Copyright

Comment (ColinSullivan1) [reply src/main/java/io/nats/client/StreamState.java:1]: Is there a corresponding change to rename StreamInfo to StreamState in the go client?  We'll want to keep the APIs as consistent as we can.

File: src/main/java/io/nats/client/StreamState.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+public class StreamState {
+    private final long msgs;
+    private final long bytes;
+    private final long firstSeq;
+    private final long lastSeq;
+    private final long consumerCount;
+    private final ZonedDateTime firstTime;
+    private final ZonedDateTime lastTime;
+
+    public StreamState(String json) {
```
Comment (ColinSullivan1) [inline src/main/java/io/nats/client/StreamState.java:18]: Does this need to be public?

**Review by ColinSullivan1 — COMMENTED** [review]
...have a few questions - minor stuff - overall LGTM!

Having trouble keeping up with how fast you churn out these PRs...  :)

File: src/examples/java/io/nats/examples/NatsJsManageStreams.java (hunk 1 above)
Comment (scottf) [reply src/examples/java/io/nats/examples/NatsJsManageStreams.java:117]: fixed

File: src/main/java/io/nats/client/impl/PurgeResponse.java (hunk 3 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/PurgeResponse.java:25]: fixed

File: src/main/java/io/nats/client/impl/PurgeResponse.java (hunk 2 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/PurgeResponse.java:30]: done. Some other place I missed too

File: src/main/java/io/nats/client/StreamState.java (hunk 5 above)
Comment (scottf) [reply src/main/java/io/nats/client/StreamState.java:1]: fixed

File: src/main/java/io/nats/client/impl/PurgeResponse.java (hunk 4 above)
Comment (scottf) [reply src/main/java/io/nats/client/impl/PurgeResponse.java:34]: done

File: src/main/java/io/nats/client/StreamState.java (hunk 6 above)
Comment (scottf) [reply src/main/java/io/nats/client/StreamState.java:18]: fixed

**Review by ColinSullivan1 — APPROVED** [review]
LGTM.  Thanks!

**Pull request merged**